extensions/README.md

```diff
@@ -0,0 +1,152 @@
+
+Once you have your custom template you can install it by just copying to ${user.home}/.blade/extensions or using the
+`$ blade extension install /path/to/org.myorg.project.templates.mywebapp-1.0.0.jar`
+
+** Give it a try! **
+```
+$ blade create -l
+```
+Now you should see your new custom project template listed!.  If not, contact us on our #blade forum on the Liferay community slack channel.  https://community.liferay.com/it/chat
+
+### Blade Profiles
+
+Lets say that you want to customize the normal blade development workflow.  Normally Liferay developers who use blade will run a series of blade commands that all make sense in the "default" lifeary workspace.
+
+1. blade init
+2. blade server init
+3. blade server start
+4. blade create my-project
+5. blade deploy
+6. blade customCommand
+
+But if this workspace wants to support `docker` or another _containerized_ Liferay environment, all of those commands
+should do something slightly different.  And the case of the last command, it is a *new* command that only makes sense
+inside of this development environment.
+
+So a blade profile allows extensions developers to "override" existing blade commands or add "new" commands.
+So in the workflow above, blade init for a profile "myprofile" would override the default "init" command do something
+before/after the normal init command.  `blade server start` in the case of a profile would override the `ServerStartCommand` in blade and contribute its own way to start the Liferay server (maybe in a container of some sort).
+
+`blade deploy` or `blade watch` would likely mean something completely different in the context of a profile, so that command
+would need to be overridden.
+
+The way to "override" existing commands can be see here in the example: https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-profile/src/main/java/com/liferay/extensions/sample/profile
+
+Here we have a "new" command that is only availble in this profile and also an "overridden" command that overrides an existing blade command just for this profle.
+
+The way that a "profile" is set, so that blade knows which profile is active depends on the following file in the base
+of your Liferay workspace
+
+`${workspaceDir}/.blade/settings.properties`  and in there, there is a property for the profile name that set.
+
+The name of a profile that is "set" on a workspace has to do with the option that you specific in the `blade init` command.
+
+```
+$ blade init -p myprofile my-new-custom-workspace
+```
+
+After that you will see there is a `my-new-custom-workspace/.blade/settings.properties` file and in there is this:
+```
+liferay.version.default=7.1
+profile.name=myprofile ```
+
+
+## Roadmap
+
+We have some existing plans for blade extensions but are looking for feedback from community of developers.
```

Comment:
"from community" I think should be "from our community", and perhaps we can add a call-to-action here to encourage the reader to give us their feedback on their blade extension development experience?